### ISSUE TITLE
[tinker][bug] `loss_fn_outputs` extraction uses left-aligned slicing on right-aligned tensors (Tinker API path)

### ISSUE THREAD

**Issue by CharlieFRuan** [NovaSky-AI/SkyRL#1304]
### Summary

`_to_training_batch()` right-aligns response-level tensors (padding on the left), but the `loss_fn_outputs` extraction in both the FSDP and Megatron workers slices with `[:valid_len]` (left-aligned), returning padding/prompt logprobs instead of the actual response logprobs.

### Root cause

[`_to_training_batch()`](https://github.com/NovaSky-AI/SkyRL/blob/954b2ee2d75fd20adf97e1729e6d1ca8342e820c/skyrl/backends/skyrl_train_backend.py#L286-L290) pads response-level tensors on the **left**:

```python
action_pad = max_response_len - len(weights)
loss_masks.append([0.0] * action_pad + [float(w) for w in weights])
response_masks.append([0] * action_pad + [1] * len(weights))
```

This `response_mask` becomes [`experience.action_mask`](https://github.com/NovaSky-AI/SkyRL/blob/954b2ee2d75fd20adf97e1729e6d1ca8342e820c/skyrl/backends/skyrl_train/workers/worker_utils.py#L83) in the worker. `max_response_len` becomes [`experience.num_actions`](https://github.com/NovaSky-AI/SkyRL/blob/954b2ee2d75fd20adf97e1729e6d1ca8342e820c/skyrl/backends/skyrl_train/workers/worker_utils.py#L84).

The model's forward pass produces `action_log_probs` by taking the [last `num_actions` logprobs](https://github.com/NovaSky-AI/SkyRL/blob/954b2ee2d75fd20adf97e1729e6d1ca8342e820c/skyrl/backends/skyrl_train/workers/model_wrapper.py#L378), so the tensor is also right-aligned — real response values on the right, prompt-region values on the left.

But the extraction code computes `valid_len` correctly via `action_mask.sum()` and then slices from the **left**:

**FSDP worker, SFT path** ([worker.py#L846-L855](https://github.com/NovaSky-AI/SkyRL/blob/954b2ee2d75fd20adf97e1729e6d1ca8342e820c/skyrl/backends/skyrl_train/workers/worker.py#L846-L855)):

```python
valid_len = int(action_mask[i].sum().item())
loss_fn_outputs.append({
    "logprobs": action_log_probs[i, :valid_len].detach().cpu().tolist(),         # BUG
    "elementwise_loss": elementwise_loss[i, :valid_len].detach().cpu().tolist(),  # BUG
})
```

**FSDP worker, RL path** ([worker.py#L900-L911](https://github.com/NovaSky-AI/SkyRL/blob/954b2ee2d75fd20adf97e1729e6d1ca8342e820c/skyrl/backends/skyrl_train/workers/worker.py#L900-L911)):

```python
valid_lens = action_mask.sum(dim=1).int().tolist()
...
    "logprobs": detached_log_probs[i, :valid_len].tolist(),  # BUG
```

**Megatron worker, SFT path** ([megatron_model_wrapper.py#L278-L289](https://github.com/NovaSky-AI/SkyRL/blob/954b2ee2d75fd20adf97e1729e6d1ca8342e820c/skyrl/backends/skyrl_train/workers/megatron/megatron_model_wrapper.py#L278-L289)):

```python
valid_len = int(action_mask[i].sum().item())
...
    "logprobs": action_log_probs[i, :valid_len].detach().cpu().tolist(),         # BUG
    "elementwise_loss": elementwise_loss[i, :valid_len].detach().cpu().tolist(),  # BUG
```

**Megatron worker, RL path** ([megatron_model_wrapper.py#L328-L342](https://github.com/NovaSky-AI/SkyRL/blob/954b2ee2d75fd20adf97e1729e6d1ca8342e820c/skyrl/backends/skyrl_train/workers/megatron/megatron_model_wrapper.py#L328-L342)):

```python
valid_lens = action_mask.sum(dim=1).int().tolist()
...
    "logprobs": detached_log_probs[i, :valid_len].tolist(),  # BUG
```

### Example

```
action_mask       = [0, 0, 1, 1, 1]
action_log_probs  = [prompt_lp, prompt_lp, real_0, real_1, real_2]
valid_len         = 3

[:3]  -> positions [0, 1, 2] -> prompt_lp, prompt_lp, real_0   (WRONG)
[-3:] -> positions [2, 3, 4] -> real_0, real_1, real_2          (CORRECT)
```

### Call path

1. [`skyrl_train_backend.py#L286-L290`](https://github.com/NovaSky-AI/SkyRL/blob/954b2ee2d75fd20adf97e1729e6d1ca8342e820c/skyrl/backends/skyrl_train_backend.py#L286-L290) — `_to_training_batch()` right-aligns `response_mask` and `loss_mask`
2. [`skyrl_train_backend.py#L313`](https://github.com/NovaSky-AI/SkyRL/blob/954b2ee2d75fd20adf97e1729e6d1ca8342e820c/skyrl/backends/skyrl_train_backend.py#L313) — stores `max_response_len` in batch metadata
3. [`skyrl_train_backend.py#L404-L409`](https://github.com/NovaSky-AI/SkyRL/blob/954b2ee2d75fd20adf97e1729e6d1ca8342e820c/skyrl/backends/skyrl_train_backend.py#L404-L409) — `forward_backward_train()` sends batch to dispatch
4. [`worker_utils.py#L83-L84`](https://github.com/NovaSky-AI/SkyRL/blob/954b2ee2d75fd20adf97e1729e6d1ca8342e820c/skyrl/backends/skyrl_train/workers/worker_utils.py#L83-L84) — `response_mask` → `experience.action_mask`, `max_response_len` → `experience.num_actions`
5. [`model_wrapper.py#L378`](https://github.com/NovaSky-AI/SkyRL/blob/954b2ee2d75fd20adf97e1729e6d1ca8342e820c/skyrl/backends/skyrl_train/workers/model_wrapper.py#L378) — `action_log_probs = log_probs[:, -num_actions - 1 : -1]` (right-aligned output)
6. `worker.py#L854-L855` / `worker.py#L911` / `megatron_model_wrapper.py#L286-L287` / `megatron_model_wrapper.py#L340` — **`[:valid_len]` slices from the wrong end**
7. [`skyrl_train_backend.py#L418-L442`](https://github.com/NovaSky-AI/SkyRL/blob/954b2ee2d75fd20adf97e1729e6d1ca8342e820c/skyrl/backends/skyrl_train_backend.py#L418-L442) — `loss_fn_outputs` packaged into `ForwardBackwardOutput` and returned to the Tinker API caller

### Impact

- The standard RL training loop [discards `loss_fn_outputs`](https://github.com/NovaSky-AI/SkyRL/blob/954b2ee2d75fd20adf97e1729e6d1ca8342e820c/skyrl/train/trainer.py#L1088), so it is **unaffected**.
- The **Tinker API path** returns corrupted `logprobs` and `elementwise_loss` values to callers — the returned arrays contain prompt-region logprobs instead of response token logprobs.
- The bug is silent: no error is raised, output shapes are correct, only the values are wrong.
- Samples whose response length equals `max_response_len` (no padding) are unaffected since `[:n]` and `[-n:]` are equivalent when `n` equals the tensor length.

### Suggested fix

Change `[:valid_len]` to `[-valid_len:]` at all four locations:

```diff
# worker.py SFT path (line 854-855)
-"logprobs": action_log_probs[i, :valid_len].detach().cpu().tolist(),
-"elementwise_loss": elementwise_loss[i, :valid_len].detach().cpu().tolist(),
+"logprobs": action_log_probs[i, -valid_len:].detach().cpu().tolist(),
+"elementwise_loss": elementwise_loss[i, -valid_len:].detach().cpu().tolist(),

# worker.py RL path (line 911)
-"logprobs": detached_log_probs[i, :valid_len].tolist(),
+"logprobs": detached_log_probs[i, -valid_len:].tolist(),

# megatron_model_wrapper.py SFT path (line 286-287)
-"logprobs": action_log_probs[i, :valid_len].detach().cpu().tolist(),
-"elementwise_loss": elementwise_loss[i, :valid_len].detach().cpu().tolist(),
+"logprobs": action_log_probs[i, -valid_len:].detach().cpu().tolist(),
+"elementwise_loss": elementwise_loss[i, -valid_len:].detach().cpu().tolist(),

# megatron_model_wrapper.py RL path (line 340)
-"logprobs": detached_log_probs[i, :valid_len].tolist(),
+"logprobs": detached_log_probs[i, -valid_len:].tolist(),
```

**Comment by tamoghnokandar** [comment]
I can take up this issue.
